<?xml version="1.0"?>

<launch>

	<!-- INPUT METHOD -->
	<arg name="SNP" default="false" />
	<arg name="KEY" default="false" />
	<arg name="JOINT" default="false"/>
	<arg name="CHIN_PORT" default="/dev/chin_switch_arduino" />
	<arg name="interface_velocity_dim" default="1"/>

	<rosparam param="snp_velocity_scale">3</rosparam>
	<rosparam param="mode_switch_paradigm">2</rosparam>
	<rosparam param="interface_dim">1</rosparam>


	
	<group if="$(arg SNP)">
		<rosparam command="load" file="$(find teleop_nodes)/config/sip_and_puff_mapping.yaml" />
		<rosparam command="load" file="$(find teleop_nodes)/config/velocity_limits.yaml" />
		<node pkg="joy" type="joy_node" name="joy_node_snp">
			<param name="dev" value="/dev/input/js1"/>
		</node>
		<node pkg="teleop_nodes" type="sip_puff_mapping.py" name="sip_puff_mapping" output="screen"/>
		<node pkg="teleop_nodes" type="sip_and_puff_hybrid_signal.py" name="sip_and_puff_hybrid_signal" output="screen" args="$(arg interface_velocity_dim)">
			<param name="snp_paradigm" value="$(find teleop_nodes)/cfg/SipPuffModeSwitchParadigm.cfg"/>
		</node>
	<!--<node pkg="rosserial_python" name="chin_button_arduino" type="serial_node.py" args="$(arg CHIN_PORT)"/>-->
	</group>


</launch>
